tests/rados/test_osd_full.py

```diff
@@ -0,0 +1,309 @@
+"""
+Module to verify scenarios related to OSD being nearfull, backfill-full
+and completely full
+"""
+import datetime
+import time
+
+from ceph.ceph_admin import CephAdmin
+from ceph.rados.core_workflows import RadosOrchestrator
+from ceph.rados.pool_workflows import PoolFunctions
+from ceph.rados.rados_bench import RadosBench
+from tests.rados.stretch_cluster import wait_for_clean_pg_sets
+from utility.log import Log
+from utility.utils import method_should_succeed
+
+log = Log(__name__)
+
+
+def run(ceph_cluster, **kw):
+    """
+    #CEPH-83571715
+    Verify cluster behaviour when OSDs are nearfull, backfill-full, and completely full.
+    Autoscale PGs, ensure proper backfilling and rebalancing
+    1. Create replicated pool with single PG and autoscaling disabled
+    2. Retrieve the acting set for the pool
+    3. Calculate the amount of data to be written to pools in
+        order to trigger near-full, backfill-full, and completely
+        full OSD health warnings
+    4. Perform IOPS using rados bench to fill up the pool
+        up till nearfull, backfillfull and completely full capacity
+    5. Verify the number of objects written to the pool and the cluster
+        health warning after each IOPS iteration
+    6. Perform IOPS again and ensure IOPS was unsuccessful as OSDs are full
+    7. Increase the set-full-ratio to a higher value and set-backfillfull-ratio
+        to a value higher than the previous value of set-full-ratio
+    8. Ensure that completely full and backfillfull warning disappear and
+        nearfull warning remains
+    9. Enable autoscaling and observe the acting set OSD's 'used capacity' decreasing
+    10. Eventually nearfull warning should disappear, wait for PGs to achieve active+clean state
+    """
+
+    log.info(run.__doc__)
+    config = kw["config"]
+    cephadm = CephAdmin(cluster=ceph_cluster, **config)
+    rados_obj = RadosOrchestrator(node=cephadm)
+    pool_obj = PoolFunctions(node=cephadm)
+    client_node = ceph_cluster.get_nodes(role="client")[0]
+    bench_obj = RadosBench(mon_node=cephadm, clients=client_node)
+    osd_sizes = {}
+    bench_obj_size_kb = 16384
```

Comment:
Just add a comment, stating the reason for choosing the above size for IOs. If any.

Reply:
Done, the size of 16 MB was chosen as official RHCS doc say that this is safe max value
Reference: Step 5 in https://access.redhat.com/documentation/en-us/red_hat_ceph_storage/6/html/administration_guide/ceph-performance-benchmark#benchmarking-ceph-performance_admin